core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ConcurrencyController.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.executor.dedicated;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+class ConcurrencyController
```

Comment:
Is this based on `io.trino.execution.SplitConcurrencyController`?

Reply:
Yes, it's the same logic as that class, but adapted to the new scheduler.